internal/api/genapi/templates.go

```diff
@@ -24,7 +24,11 @@ func getArgsAndPaths(in []string, parentTypeName, action string) (colArg, resArg
 		strToReplace = in[len(in)-2]
 	}
 	colArg = fmt.Sprintf("%sId", strcase.ToLowerCamel(strings.ReplaceAll(strToReplace, "-", "_")))
-	colPath = fmt.Sprintf("%ss", in[len(in)-1])
+	colPath = in[len(in)-1]
```

Comment:
This change is unnecessary. The problem it's trying to fix is caused because the values in input.go are incorrect. It needs to be fixed there, not here.

Reply:
@talanknight

Reply:
But librarie and librarys aren't words.

Reply:
Fair point. Can we instead allow providing a plural override into the input structs and use that directly in the template?

Reply:
PathArgs is actually an outdated field anyways which seems to date back to when we had multiple resource names in our path.  I'll rework this a bit to have Read/Update and custom actions use just a variable named `id` and replace this field to just be the plural resource name since it will only then be used for building the resource paths.